src/components/dataset/query/sidebar/panels/CreateSnapshotPanel.jsx

```diff
@@ -23,40 +22,60 @@ const styles = theme => ({
   cancelButton: {
     alignSelf: 'flex-end',
   },
+  textField: {
+    backgroundColor: theme.palette.common.white,
+    borderRadius: '5px',
+    marginBottom: theme.spacing(1),
+  },
 });
 
-export class CreateSnapshotPanel extends React.PureComponent {
-  static propTypes = {
-    classes: PropTypes.object,
-    handleCreateSnapshot: PropTypes.func,
-    handleSaveSnapshot: PropTypes.func,
-  };
-
-  render() {
-    const { classes, handleCreateSnapshot, handleSaveSnapshot } = this.props;
+function CreateSnapshotPanel(props) {
```

Comment:
Why the switch to a functional component instead of class-based? It's good for us to remain consistent if we can.

Reply:
I thought since the component doesn't do much in terms of changing the state, it would be cleaner and more concise to use the `useState` hook (I liked how it could be called directly from `onChange`). It would be fine to revert back to a class component; I would just have to write some additional functions.